### ISSUE TITLE
Using `$this`

### ISSUE THREAD

**Issue by blainesch** [bobthecow/psysh#140]
So I tried using `$this` and it didn't seem to work. Am I doing something wrong?

Using psysh @ 0fc04a5d68ae53f496d45e2add601e2eac9cf4dd
~~~
-> % php --version
PHP 5.4.29 (cli) (built: Jun  5 2014 17:58:41)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.4.0, Copyright (c) 1998-2014 Zend Technologies
~~~

**Comment by bobthecow** [comment]
`$this` is weird, you're usually better off assigning it to a different variable, unfortunately :-/

How are you getting into psysh?

**Comment by blainesch** [comment]
In code using the eval method: `eval(\Psy\sh())`

**Comment by bobthecow** [comment]
The eval method doesn't support this, unfortunately, so you'd need to use the longer, uglier invocation.

I just pushed a change to add `$this` support to the eval method, though. Check out `develop` :)

**Comment by bobthecow** [comment]
This has shipped in `v0.3.3`, and binding `$this` is even now mentioned in Usage! http://psysh.org/#usage
